pkg/yaml/types_test.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2017 The Kedge Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yaml
+
+import (
+	"encoding/json"
+	"reflect"
+	"testing"
+)
+
+type SliceArrayorIntArrayHolder struct {
+	SliceorArray SliceArrayorIntArray `json:"portMappings"`
+}
+
+func TestSliceArrayorIntArrayUnmarshalJSON(t *testing.T) {
+	cases := []struct {
+		input  string
+		result SliceArrayorIntArray
+	}{
+		{"{\"portMappings\":[3306]}", SliceArrayorIntArray([]string{"3306"})},
```

Comment:
JIC: if you don't want to escape the double quotes, then you can use the back ticks to enclose the entire string like following:

https://github.com/kedgeproject/kedge/blob/b7154f7765c89d0f29c3881e70ccc187115ad4a9/pkg/spec/controller_test.go#L155-L162